ivadomed/config/config.json

```diff
@@ -1,90 +1,109 @@
-    "batch_size": 18,
-    "dropout_rate": 0.3,
-    "batch_norm_momentum": 0.1,
-    "num_epochs": 100,
-    "initial_lr": 0.001,
-    "lr_scheduler": {"name": "CosineAnnealingWarmRestarts", "T_0": 10, "T_mult": 1},
-    "loss": {"name": "dice"},
-    "log_directory": "log_sc",
-    "film_layers": [0, 0, 0, 0, 0, 0, 0, 0],
-    "mixup_bool": false,
-    "mixup_alpha": 2,
-    "metadata": "contrast",
-    "multichannel": false,
-    "out_channel": 1,
-    "depth": 3,
-    "train_fraction": 0.6,
-    "test_fraction": 0.2,
-    "early_stopping_patience": 10,
-    "early_stopping_epsilon": 0.001,
-    "slice_axis": "axial",
-    "balance_samples": false,
-    "retrain_model": null,
-    "retrain_fraction": 1.0,
-    "split_method": "per_center",
-    "slice_filter": {"filter_empty_mask": false, "filter_empty_input": true},
-    "slice_filter_roi": 10,
-    "attention_unet": false,
-    "uncertainty": {"epistemic": true, "aleatoric": false, "n_it": 10},
-    "unet_3D": false,
-    "HeMIS": false,
-    "binarize_prediction": true,
-    "eval_params": {},
-    "transformation_training": {
-        "Resample": {
+    "log_directory": "spineGeneric",
+    "target_suffix": ["_seg-manual"],
+    "debugging": false,
+    "loader_parameters": {
+        "roi_params": {
+            "suffix": null,
+            "slice_filter_roi": null
+        },
+        "contrast_params": {
+            "training_validation": ["T1w", "T2w", "T2star", "acq-MToff_MTS", "acq-MTon_MTS", "acq-T1w_MTS"],
+            "testing": ["T1w", "T2w", "T2star", "acq-MToff_MTS", "acq-MTon_MTS", "acq-T1w_MTS"],
+            "balance": {}
+        },
+        "slice_filter_params": {
+            "filter_empty_mask": false,
+            "filter_empty_input": true
+        },
+        "slice_axis": "axial",
+        "multichannel": false
+    },
+    "split_dataset": {
+        "fname_split": null,
+        "random_seed": 6,
+        "center_test": ["stanford", "amu", "queensland", "cardiff"],
+        "method": "per_center",
+        "train_fraction": 0.6,
+        "test_fraction": 0.2
+    },
+    "training_parameters": {
+        "batch_size": 18,
+        "loss": {
+            "name": "DiceLoss"
+        },
+        "training_time": {
+            "num_epochs": 100,
+            "early_stopping_patience": 50,
+            "early_stopping_epsilon": 0.001
+        },
+        "scheduler": {
+            "initial_lr": 0.001,
+            "lr_scheduler": {
+                "name": "CosineAnnealingLR",
+                "base_lr": 1e-5,
+                "max_lr": 1e-2
+            }
+        },
+        "balance_samples": false,
+        "mixup_alpha": null,
+        "transfer_learning": {
+            "retrain_model": null,
+            "retrain_fraction": 1.0
+        }
+    },
+    "default_model": {
+        "name": "Unet",
+        "dropout_rate": 0.3,
+        "bn_momentum": 0.9,
+        "depth": 4,
+        "in_channel": 1,
```

Comment:
The number of input and output channels of a model can be calculated automatically. If multichannel is True, in_channel=len(contrast_lst) else 1. Number of output channels can be found from the target_suffix list. Adding this parameter to the config file forces the user to change more that one parameter if he wants to go from single channel to multichannel, for instance. I'd completely remove these params from the config (in_channel and out_channel) files and compute the number directly in the code.

Reply:
Makes sense! Happy to change this :-)

Reply:
done